src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs

```diff
@@ -81,7 +76,15 @@ private static ThreeState GetObsoleteContextState(Symbol symbol, bool forceCompl
                     return state;
                 }
 
-                symbol = symbol.ContainingSymbol;
+                // For property or event accessors, check the associated property or event next.
+                if (symbol.IsAccessor())
```

Comment:
This suppresses warnings/errors from the usage of an obsolete symbol in an accessor if the accessor is marked obsolete itself.

We only really need to do this for properties, not for events, but it is simplest to treat them both equally, and it is arguable which way is most correct.

For a similar reason, we change this for all C# versions, not just for C# 8